File: src/ast/ElaboratedType.java
package ast;

import java.io.StringWriter;

public class ElaboratedType extends ParentNode {
    private String _elaborated;

    public ElaboratedType(String elaborated, BaseNode type) {
        super(NodeType.ElaboratedType, type);
        _elaborated = elaborated;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write(_elaborated);
        writer.write(" ");
        child.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/ElaboratedType.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class ElaboratedType(private val elaborated: String, type: BaseNode?) : ParentNode(NodeType.ElaboratedType, type) {
    public override fun printLeft(writer: StringWriter) {
        writer.write(elaborated)
        writer.write(" ")
        child!!.print(writer)
    }

}